docs/adr/001-provisioning.md

```diff
@@ -0,0 +1,346 @@
+# Context
+This document defines the architecture and API for the Gardener cluster provisioning functionality.
+
+# Status
+Proposed
+
+# Decision
+
+The following diagram shows the proposed architecture:
+![](./assets/keb-kim-target-arch.drawio.svg)
+
+> Note: At the time of writing, the GardenerCluster CR was used to generate kubeconfig. The [workplan](https://github.com/kyma-project/infrastructure-manager/issues/112) for delivering provisioning functionality includes renaming the CR to maintain consistency.
+
+The following assumptions were taken:
+- Kyma Environment Broker must not contain all the details of the cluster infrastructure.
+- Kyma Infrastructure Manager's API must expose properties that:
+  - can be set in the BTP cockpit by the user
+  - are directly related to plans in the KEB
+- Kyma Infrastructure Manager's API must not expose properties that are:
+  - hardcoded in the Provisioner, or the KEB
+  - statically configured in the management-plane-config
+
+Kyma Environment Broker has the following responsibilities:  
+- Create Runtime CR containing the following data:
+    - Provider config (type, region, and secret with credentials for hyperscaler)
+    - Worker pool specification
+    - Cluster networking settings (nodes, pods, and services API ranges)
+    - OIDC settings
+    - Cluster administrators list
+    - Egress network filter settings
+    - Control Plane failure tolerance config
+- Observe the status of the CR to determine whether provisioning succeeded
+
+ Kyma Infrastructure Manager has the following responsibilities:
+- Create shoots based on:
+   - Corresponding `Runtime` CR properties
+   - Corresponding `Runtime` CR labels:
+     -  `kyma-project.io/platform-region` for determining if the cluster is located in EU 
+   - Predefined defaults for the optional properties:
+     - Kubernetes version
+     - Machine image version
+   - Predefined configuration for the following functionalities:
+     - configuring DNS extension 
+     - configuring Certificates extension
+     - providing maintenance settings (Kubernetes, and image autoupdates)
+     - creating provider specific config
+ - Upgrade and delete shoots for the corresponding `Runtime` CRs
+ - Apply the audit log configuration on the shoot resource
+ - Create cluster role bindings for administrators
+ - Generate the kubeconfig
+
+## API proposal
+
+### CR examples
+
+MInd that the Runtime CR must contain the following labels:
```

Comment:
Why these labels are required? Is it for KEB needed or the operator which tries to find instances?

Reply:
> Why these labels are required? Is it for KEB needed or the operator which tries to find instances?

The labels are used by KIM operator to annotate created shoot (e.g. to determine EU access)

The labels may be also useful for on-call related activities to e.g. search for Runtime CR for particular subaccount. Please also mind the Kyma resource have the same set of labels.

Reply:
I find it strange that we put so much data into the labels instead providing them as part of the API. If they are required to let KIM do the proper job, why they are not mandatory fields in the API? The labels should be used for additional search use cases only I assume.

I assume that this is all historical as KEB stores its data in a database and does not manage a well-defined k8s resource containing the BTP data. So you cannot just reference that entity and need to duplicate it into all resources to make thinks findable. Best would be to search for a BTP entity, so that you can then navigate to the referenced shoot resource.

Maybe we still can improve something here and have mandatory data modelled in the API properly?

Reply:
> If they are required to let KIM do the proper job, why they are not mandatory fields in the API?

Good point. I think it makes sense to  remove `kyma-project.io/platform-region` label and add `spec.shoot.platformRegion` property. 

Other fields (e.g. `subaccount-id`) are used for searching purpose. They may be also be used for annotating shoot resource (actually it's not decided what annotations will be added), but I don't consider them as a part of contract (if the labels will be skipped, KIM will still be able to create cluster for all scenarios).

I will also rephrase the sentence: "Mind that the Runtime CR must contain the following labels" to stress that the label are not required, but will be helpful for efficient searching.